tests/unit/algorithms/anomaly/adapters/callbacks/test_inference_callback.py

```diff
@@ -0,0 +1,26 @@
+"""Tests the inference callback on a dummy lightning module."""
+
+# Copyright (C) 2021-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import pytorch_lightning as pl
+
+from otx.algorithms.anomaly.adapters.anomalib.callbacks import AnomalyInferenceCallback
+from otx.api.entities.model_template import TaskType
+from tests.unit.algorithms.anomaly.helpers.dummy_dataset import MockDataModule
+from tests.unit.algorithms.anomaly.helpers.dummy_model import DummyModel
+
+
+class TestInferenceCallback:
+    @pytest.mark.parametrize(
+        "task_type", [TaskType.ANOMALY_CLASSIFICATION, TaskType.ANOMALY_DETECTION, TaskType.ANOMALY_SEGMENTATION]
+    )
+    def test_inference_callback(self, task_type):
```

Comment:
As discussed, we might want to add more tests to cover various use cases and edge cases (possibly in a separate PR). Some suggestion:
- all three task types (cls, det, seg)
- Are heatmaps stored correctly as metadata items?